src/libsmac/src/parser.rs

```diff
@@ -4,7 +4,7 @@ use std::str::from_utf8;
 
 // Identifier
 // IDENT := /[a-zA-Z_]\w*/g
-named!(ident<&str, &str>, re_match!(r"[a-zA-Z_]\w*"));
+named!(ident<&str, &str>, re_match!(r"[a-zA-Z_]\w*[\?\!]*"));
```

Comment:
Should we only allow `!` and `?` at the end or also inside of the identifier?
We could use `(?:\w|[\?\!])*` instead to allow it everywhere.
Do you think we should allow that or not? I'm not sure if it's needed.